### PR TITLE
Support -T rootsig_1_0/rootsig_1_1.

### PR DESCRIPTION
**Pull request by python3kgae**

### CODE REVIEW

File: tools/clang/tools/dxcompiler/dxcompilerobj.cpp [hunk 1]
```diff
@@ -2081,6 +2092,38 @@ class DxcCompiler : public IDxcCompiler, public IDxcLangExtensions, public IDxcC
...
+          IFT(CreateMemoryStream(pMalloc, &pContainerStream));
+          SerializeDxilContainerForRootSignature(rootSigHandle.get(),
+                                                 pContainerStream);
+
+          pOutputBlob.Release();
+          IFT(pContainerStream.QueryInterface(&pOutputBlob));
+
+          CComPtr<IDxcBlob> pTargetBlob;
+          if (m_pDxcContainerEventsHandler != nullptr) {
+            HRESULT hr = m_pDxcContainerEventsHandler->OnDxilContainerBuilt(
```
Comment (dmpots) [inline tools/clang/tools/dxcompiler/dxcompilerobj.cpp]: Probably do not need to call the event here. I think it was intended as a callback after compilation succeeded.

File: tools/clang/unittests/HLSL/CompilerTest.cpp [hunk 2]
```diff
@@ -591,6 +591,8 @@ class CompilerTest {
   TEST_METHOD(CodeGenResPhi)
   TEST_METHOD(CodeGenResPhi2)
   TEST_METHOD(CodeGenRootSigEntry)
+  TEST_METHOD(CodeGenRootSigProfile)
+  TEST_METHOD(CodeGenRootSigProfile2)
```
Comment (dmpots) [inline tools/clang/unittests/HLSL/CompilerTest.cpp]: Should add tests for the two negative cases we support:

1. Root signature macro name is not found.
2. Root signature in macro is invalid.
